stl/inc/vector

```diff
@@ -441,6 +441,9 @@ private:
     friend class _Vb_val;
     friend _Tidy_guard<vector>;
 
+    template <class>
+    friend class basic_stacktrace;

```

Comment:
I believe we should guard this with `_HAS_CXX23` (otherwise, `basic_stacktrace` is effectively made reserved in pre-C++23 modes).
```suggestion
#if _HAS_CXX23
    template <class>
    friend class basic_stacktrace;
#endif // _HAS_CXX23
```